src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -363,13 +364,60 @@ public void run(final RunNotifier notifier) {
             statement.evaluate();
         } catch (AssumptionViolatedException e) {
             testNotifier.addFailedAssumption(e);
-        } catch (StoppedByUserException e) {
-            throw e;
         } catch (Throwable e) {
-            testNotifier.addFailure(e);
```

Comment:
I have a simpler suggestion that still works. We could add a method to `RunNotifier` called `stopRequested()` that returns `true` if `pleaseStop()` was called. Right here, after calling `addFailure()`, we would call that method, and if it returns `true` throw `StoppedByUserException`

Note that in #1118 someone is proposing adding `fireTestSuiteStarted()` and `fireTestSuiteFinished()` to `RunNotifier`. Those methods would be other good places to throw `StoppedByUserException()`